src/emu/sound.cpp

```diff
@@ -579,6 +580,7 @@ sound_stream::sound_stream(device_t &device, u32 inputs, u32 outputs, u32 output
 	std::string state_tag = string_format("%d", m_device.machine().sound().unique_id());
 	auto &save = m_device.machine().save();
 	save.save_item(&m_device, "stream.sample_rate", state_tag.c_str(), 0, NAME(m_sample_rate));
+	save.save_item(&m_device, "stream.sample_rate", state_tag.c_str(), 0, NAME(m_last_update_end));
```

Comment:
Same name?  How the hell did that work?

Reply:
I assume it's `NAME(m_last_update_end)` as the last parameter. I'll fix that.

Reply:
Here is another one:
```
void sound_stream_input::init(sound_stream &stream, u32 index, char const *tag, sound_stream_output *resampler)
{
	// set the passed-in values
	m_owner = &stream;
	m_index = index;
	m_resampler_source = resampler;

	// save our state
	auto &save = stream.device().machine().save();
	save.save_item(&stream.device(), "stream.input", tag, index, NAME(m_gain));
	save.save_item(&stream.device(), "stream.input", tag, index, NAME(m_user_gain));
}
```
There it more looks like that the code should be something like
```
	save.save_item(&m_device, "stream.base_stream", state_tag.c_str(), 0, NAME(m_sample_rate));
	save.save_item(&m_device, "stream.base_stream", state_tag.c_str(), 0, NAME(m_last_update_end));
```